### PR TITLE
Add Int#bit_length

### PR DESCRIPTION
**Pull request by asterite**
Will be useful to implement https://forum.crystal-lang.org/t/integer-sqrt-in-stdlib . It's probably also a generally useful method. Also available in Ruby.

### CODE REVIEW

File: src/int.cr [hunk 1]
```diff
@@ -395,6 +395,36 @@ struct Int
...
   end
 
+  # Returns the number of bits of this int value.
+  #
+  # “The number of bits” means that the bit position of the highest bit
+  # which is different to the sign bit.
+  # (The bit position of the bit 2**n is n+1.)
+  # If there is no such bit (zero or minus one), zero is returned.
+  #
+  # I.e. This method returns `ceil(log2(self < 0 ? -self : self+1))`.
```
Comment (vlazar) [inline src/int.cr]: Example formatted with `crystal tool format`
```suggestion
  # I.e. This method returns `ceil(log2(self < 0 ? -self : self + 1))`.
```

Comment (asterite) [reply src/int.cr]: I copied this verbatim from Ruby's documentation. But I guess this is fine. Thanks!

File: spec/std/int_spec.cr [hunk 2]
```diff
@@ -777,4 +777,11 @@ describe "Int" do
     65.unsafe_chr.should eq('A')
     (0x10ffff + 1).unsafe_chr.ord.should eq(0x10ffff + 1)
   end
+
+  it "#bit_length" do
```
Comment (RX14) [inline spec/std/int_spec.cr]: Test negative numbers? perhaps `BigInt` too but that's not neccesary if it doesn't work (but you should move it to only primitive ints if it doesn't)

File: spec/std/int_spec.cr [hunk 3]
```diff
@@ -777,4 +777,11 @@ describe "Int" do
...
+    0.bit_length.should eq(0)
+    (1..100).each do |i|
+      i.bit_length.should eq(Math.log2(i).to_i + 1)
```
Comment (oprypin) [inline spec/std/int_spec.cr]: This is testing that the result of `bit_length` matches some mathematical formula. But it has about as much to do with the expected answers as the actual implementation of the method. Should you then have to write tests for this mathematical formula?
Tests shouldn't have logic, I'd suggest checking actual inputs.

Comment (oprypin) [reply spec/std/int_spec.cr]: Things like `0b1000` and `0b111.bit_length` might look nice

File: spec/std/int_spec.cr (hunk 2 above)
Comment (asterite) [reply spec/std/int_spec.cr]: I added one for negative numbers. I also added a fallback for non Int::Primitive types.
Thanks!

File: spec/std/int_spec.cr (hunk 3 above)
Comment (asterite) [reply spec/std/int_spec.cr]: Good idea. Done!

Comment (oprypin) [reply spec/std/int_spec.cr]: Nice, thanks. But I think it still needs some checks specifically with "1+all-zeros" (like `0b1000`) and "all-ones" (`0b111`) to catch off-by-one errors.

Comment (asterite) [reply spec/std/int_spec.cr]: How about now?

File: spec/std/int_spec.cr [hunk 4]
```diff
@@ -1,4 +1,4 @@
-require "spec"
+require "./spec_helper"
```
Comment (straight-shoota) [inline spec/std/int_spec.cr]: Why do we need `spec_helper`?

Comment (asterite) [reply spec/std/int_spec.cr]: Good question. The code in master doesn't compile if you run just that spec file, it's missing `pending_win32` or something.

Comment (Sija) [reply spec/std/int_spec.cr]: This change shouldn't be needed anymore.

Comment (asterite) [reply spec/std/int_spec.cr]: Why not?

**Pull request merged**